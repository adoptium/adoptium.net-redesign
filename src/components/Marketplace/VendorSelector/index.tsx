import React, { useEffect, useState } from "react"
import vendors from "../../../json/marketplace.json"
import getVendorIdentifier from "../../../util/vendors"
import { shuffle } from "../../../util/shuffle"

export interface Vendor {
  name: string
  icon: string
  iconPadding?: string
}

interface VendorSelectorProps {
  selectedVendorIdentifiers: string[]
  setSelectedVendorIdentifiers: (identifiers: string[]) => void
}

const VendorSelector: React.FC<VendorSelectorProps> = ({
  selectedVendorIdentifiers,
  setSelectedVendorIdentifiers,
}) => {
  const [randomizedVendors, setRandomizedVendors] = useState<Vendor[]>([])

  useEffect(() => {
    const vendorsCpy = [...vendors] // Assuming `vendors` is typed correctly
    setRandomizedVendors(shuffle(vendorsCpy))
    setSelectedVendorIdentifiers(
      vendorsCpy.map(vendor => getVendorIdentifier(vendor)),
    )
  }, [setSelectedVendorIdentifiers])

  const handleChange = (
    e: React.MouseEvent<HTMLLIElement, MouseEvent>,
    identifier: string,
  ) => {
    e.preventDefault()
    const newSelectedVendorIdentifiers = [...selectedVendorIdentifiers]
    const idx = newSelectedVendorIdentifiers.indexOf(identifier)
    if (idx >= 0) newSelectedVendorIdentifiers.splice(idx, 1)
    else newSelectedVendorIdentifiers.push(identifier)
    setSelectedVendorIdentifiers(newSelectedVendorIdentifiers)
  }

  return (
    <div className="w-full">
      <div className="overflow-auto min-w-full w-full">
        <div className="flex xl:justify-center space-x-5 py-2">
          {randomizedVendors.map(vendor => {
            const identifier = getVendorIdentifier(vendor)
            return (
              <button
                key={`vendor-${identifier}`}
                onClick={e => handleChange(e, identifier)}
                className={`flex justify-center items-center
                outline-hidden cursor-pointer transition-all duration-200 ease-in-out bg-white rounded-2xl ${selectedVendorIdentifiers.includes(identifier) ? "border-primary border-2" : "border-white border-2"}`}
              >
                <span className="w-20 h-20">
                  <img
                    src={`/images/vendors/${vendor.icon}`}
                    className="mb-0 p-1"
                    alt={`${vendor.name} icon`}
                    style={
                      vendor.iconPadding ? { padding: vendor.iconPadding } : {}
                    }
                  />
                </span>
              </button>
            )
          })}
        </div>
      </div>
    </div>
  )
}

export default VendorSelector
