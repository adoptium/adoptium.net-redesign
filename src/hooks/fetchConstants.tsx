import { useEffect, useState } from 'react';
import axios from 'axios';

const baseUrl = 'https://api.adoptium.net/v3';

export function fetchOses(isVisible: boolean): OperatingSystem[] {

    const [oses, setOses] = useState<OperatingSystem[]>([]);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/types/operating_systems`;

            axios.get(url)
                .then(function (response) {
                    const newOses = response.data.map(s => {
                        const o: OperatingSystem = { name: s, value: s.toLowerCase() }
                        return o
                    })

                    setOses(newOses);
                })
                .catch(function (error) {
                    setOses([]);
                });
        })();
        }
    }, [isVisible]);

    return oses;
}

export function fetchArches(isVisible: boolean): Architecture[] {

    const [arches, setArches] = useState<Architecture[]>([]);

    useEffect(() => {
        if (isVisible) {
        (async () => {
            const url = `${baseUrl}/types/architectures`;

            axios.get(url)
                .then(function (response) {
                    const newArches = response.data.map(s => {
                        const a: Architecture = { name: s, value: s.toLowerCase() }
                        if(a.name === 'x32') a.name = 'x86'
                        return a;
                    })

                    setArches(newArches);
                })
                .catch(function (error) {
                    setArches([]);
                });
        })();
        }
    }, [isVisible]);

    return arches;
}

export interface OperatingSystem {
    name: string
    value: string
}

export interface Architecture {
    name: string
    value: string
}
