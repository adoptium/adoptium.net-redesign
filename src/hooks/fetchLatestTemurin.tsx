import { useEffect, useState } from "react"
import { VersionMetaData } from "."
import axios from "axios"

const baseUrl = "https://api.adoptium.net/v3"

export function fetchLatestForOS(
  isVisible: boolean,
  version: number,
  os: string,
  arch: string,
): Binary | null {
  if (!os) {
    return null
  }

  const [binary, setBinary] = useState<Binary | null>(null)

  useEffect(() => {
    if (isVisible) {
      (async () => {
        const url = `${baseUrl}/assets/feature_releases/${version}/ga?os=${os}&architecture=${arch}&image_type=jdk&jvm_impl=hotspot&page_size=1&vendor=eclipse`

        axios
          .get(url)
          .then(function (response) {
            const json: LatestTemurin = response.data[0]

            let binary_link = json.binaries[0].package.link
            let binary_checksum = json.binaries[0].package.checksum

            if (json.binaries[0].installer) {
              ;(binary_link = json.binaries[0].installer.link),
                (binary_checksum = json.binaries[0].installer.checksum)
            }

            const binary = {
              release_name: json.release_name,
              link: binary_link,
              checksum: binary_checksum,
            }
            setBinary(binary)
          })
          .catch(function (error) {
            setBinary(null)
          })
      })()
    }
  }, [isVisible, version, os, arch])

  return binary
}

export interface Binary {
  release_name: string
  link: URL
  checksum: string
}

export interface LatestTemurin {
  download_count: number
  id: string
  release_link: URL
  release_name: string
  release_type: string
  source?: {
    link: URL
    name: string
    size: number
  }
  timestamp: Date
  updated_at: Date
  vendor: string
  version_data: VersionMetaData
  binaries: [
    {
      architecture: string
      download_count: number
      heap_size: string
      image_type: string
      jvm_impl: string
      os: string
      package: {
        checksum: string
        checksum_link: URL
        download_count: number
        link: URL
        metadata_link: URL
        name: string
        signature_link: URL
        size: number
      }
      installer?: {
        checksum: string
        checksum_link: URL
        download_count: number
        link: URL
        metadata_link: URL
        name: string
        signature_link: URL
        size: number
      }
    },
  ]
}
