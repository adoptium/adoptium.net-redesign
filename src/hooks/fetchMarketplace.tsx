import { VersionMetaData } from "."
import axios from "axios"

const baseUrl = "https://marketplace-api.adoptium.net"

export async function getAllPkgsForVersion(
  version: number,
  os: string,
  architecture: string,
  package_type: string,
  vendors: string[] = [],
): Promise<MarketplaceRelease[] | null> {
  let params = "?"
  params += "feature_version=" + version

  if (os !== "any") {
    if (os === "alpine-linux") {
      params += "&os=alpine_linux"
    } else {
      params += "&os=" + os
    }
  }

  // loop through vendors and add to params
  if (vendors.length > 0) {
    vendors.forEach(vendor => {
      params += "&vendor=" + vendor
    })
  }

  if (architecture !== "any") {
    params += "&architecture=" + architecture
  }

  const url = baseUrl + "/v1/assets/latestForVendors" + params
  const data = await axios
    .get(url)
    .then(function (response) {
      return response.data
    })
    .catch(function (error) {
      return null
    })

  return data
}

export function getImageForDistribution(distribution: string) {
  switch (distribution) {
    case "temurin":
      return "/images/adoptium-logo.png"
    case "redhat":
      return "/images/redhat.svg"
    case "bisheng":
      return "/images/huawei.svg"
    case "zulu":
      return "/images/azul-logo.png"
    case "semeru":
      return "/images/ibm-logo.png"
    default:
      return `/images/${distribution}-logo.png`
  }
}

export interface MarketplaceRelease {
  release_name: string
  vendor: string
  binary: {
    os: string
    architecture: string
    image_type: string
    jvm_impl: string
    package: {
      name: string
      link: URL
      sha265sum: string
      sha256sum_link: URL
      signature_link: URL
    }
    installer?: [
      {
        name: string
        link: URL
        sha265sum: string
        sha256sum_link: URL
        signature_link: URL
      },
    ]
    timestamp: Date
    scm_ref: string
    openjdk_scm_ref: string
    distribution: string
    aqavit_results_link: URL
    tck_affidavit_link: URL
  }
  version: VersionMetaData
}
