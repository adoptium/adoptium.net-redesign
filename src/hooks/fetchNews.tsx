import { useEffect, useState } from "react"
import axios from "axios"

const baseUrl = "https://newsroom.eclipse.org/api"

export function fetchNewsItems(isVisible: boolean, page: number): News | null {
  const [news, setNews] = useState<NewsResponse>({ news: [], pagination: null })
  const [events, setEvents] = useState<EventItem[]>([])

  useEffect(() => {
    if (isVisible) {
      (async () => {
        setNews(await fetchLatestNews(page))
        setEvents(await fetchLatestEvents())
      })()
    }
  }, [isVisible, page])

  const newsAndEvents: News = {
    news: news,
    events: events,
  }

  return newsAndEvents
}

async function fetchLatestNews(page) {
  const url = new URL(`${baseUrl}/news`)
  url.searchParams.append("parameters[publish_to]", "adoptium")
  url.searchParams.append("page", page)
  url.searchParams.append("pagesize", "5")

  return axios
    .get(url.toString())
    .then(function (response) {
      return response.data
    })
    .catch(function (error) {
      return []
    })
}

async function fetchLatestEvents() {
  const url = new URL(`${baseUrl}/events`)
  url.searchParams.append("parameters[publish_to]", "adoptium")

  return axios
    .get(url.toString())
    .then(function (response) {
      return response.data.events
    })
    .catch(function (error) {
      return []
    })
}

export interface News {
  news: NewsResponse
  events: EventItem[]
}

export interface NewsResponse {
  news: NewsItem[]
  pagination: {
    page: number
    pagesize: number
    result_start: number
    result_end: number
    result_size: number
    total_result_size: number
  } | null
}

export interface NewsItem {
  id: string
  title: string
  body: string
  date: Date
  link: URL
}

export interface EventItem {
  id: string
  title: string
  infoLink: URL
  date: Date
}

export interface EventAPI {
  events: EventItem[]
}
