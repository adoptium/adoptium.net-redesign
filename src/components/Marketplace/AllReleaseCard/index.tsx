import React from "react"
import { Link } from "gatsby-plugin-react-i18next"
import { FaApple } from "react-icons/fa"

import { capitalize } from "../../../util/capitalize"
import { getImageForDistribution } from "../../../hooks"

const AllReleaseCard = ({ results }) => {
  return (
    <div className="mt-8">
      {results.map((release, index) => (
        <div
          key={index}
          className="newscard-2 !blur-0 px-6 py-6 lg:flex  justify-between items-center mt-6"
        >
          <div className="flex flex-col lg:flex-row justify-between gap-4 lg:items-center">
            <span className="block md:hidden">
              {getImageForDistribution(release.vendor)}
            </span>
            <div className="lg:w-[160px] mt-3 md:mt-0">
              <h3 className="text-[24px] hidden md:block md:text-[16px] font-bold leading-[133.333%] md:leading-[150%]">
                {release.release_name} - {release.vendor}
              </h3>
              <h3 className="text-[24px] block md:hidden md:text-[16px] font-bold leading-[133.333%] md:leading-[150%]">
                {release.release_name} - {release.vendor}
              </h3>
            </div>
            <div className="lg:flex  items-center gap-6">
              <p className="text-base text-[#C4BFCE] leading-[150%] mb-0  hidden lg:block">
                Release Date:{" "}
                {new Date(release.binary.timestamp).toLocaleDateString()}
              </p>
              <span className="fill-primary hidden md:block">
                <img
                  className="w-[100px] mb-0"
                  alt={`${release.binary.distribution} logo`}
                  src={getImageForDistribution(release.binary.distribution)}
                />
              </span>
              <p className="text-base text-[#C4BFCE] leading-[150%] hidden lg:block mb-0">
                {capitalize(release.binary.os)}
              </p>
              <p className="text-base text-[#C4BFCE] leading-[150%] mb-0 hidden lg:block">
                {release.binary.architecture}
              </p>
            </div>
          </div>
          <div className="flex items-end justify-between mt-4 md:mt-0 gap-5">
            <Link
              placeholder="Download"
              to={release.binary.package.link}
              className="rounded-[80px] hover:shadow-2xl transition-all duration-300 bg-[#FF1464] border ease-in-out border-[#FF1464] flex items-center justify-center gap-3 px-8 py-4 text-white font-bold leading-6 text-base "
            >
              <span>
                <FaApple />
              </span>
              Download
            </Link>
          </div>
        </div>
      ))}
    </div>
  )
}

export default AllReleaseCard
