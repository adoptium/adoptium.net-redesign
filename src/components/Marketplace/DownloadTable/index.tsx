import React, { useEffect, useState } from "react"
import { useStaticQuery, graphql } from "gatsby"
import { Trans } from "gatsby-plugin-react-i18next"
import { useLocation } from "@gatsbyjs/reach-router"
import queryString from "query-string"

import ReleaseSelector from "../../Temurin/ReleaseSelector"
import VendorSelector from "../VendorSelector"
import AllReleaseCard from "../AllReleaseCard"

import { setURLParam } from "../../../util/setURLParam"
import { detectOS, UserOS } from "../../../util/detectOS"
import { getAllPkgsForVersion, MarketplaceRelease } from "../../../hooks"
import { fetchOses, fetchArches} from '../../../hooks/fetchConstants'
import { packageTypes, defaultArchitecture, defaultPackageType} from '../../../util/defaults'

const DownloadTable = () => {

  const data = useStaticQuery(graphql`
    query MarketplaceVersionsQuery {
      allVersions(sort: { version: DESC }, filter: { lts: { eq: true } }) {
        edges {
          node {
            version
            label
            lts
          }
        }
      }
      mostRecentLts {
        version
      }
    }
  `)

  // load OS and Arches from API
  const oses = fetchOses(true)
  const arches = fetchArches(true)
  const [ready, setReady] = useState(false)

  const queryStringParams = queryString.parse(useLocation().search)

  const [os, updateOS] = useState("")
  const [arch, updateArch] = useState(defaultArchitecture)
  const [version, udateVersion] = useState(data.mostRecentLts.version)
  const [packageType, updatePackageType] = useState(defaultPackageType)

  const [selectedVendorIdentifiers, updateSelectedVendorIdentifiers] = useState<string[]>([])

  const [releases, setReleases] = useState<MarketplaceRelease[] | null>(null)

  /**
   * This useEffect() is called when OS and arches are finaly loaded
   */ 
  useEffect(() => {
    // do nothing while OS and arches are not loaded
    if(oses.length === 0 || arches.length === 0) return;

    (async () => {
      // init the default selected Architecture, if any from the param 'arch'
      let defaultSelectedArch = defaultArchitecture
      const archParam = queryStringParams.arch
      if (archParam) {
        let archParamStr = archParam.toString().toLowerCase()
        if (arches.findIndex(a => a.value === archParamStr) >= 0)
          defaultSelectedArch = archParamStr
      }

      // init the default selected Operation System, if any from the param 'os'
      let defaultSelectedOS = ""
      const osParam = queryStringParams.os
      if (osParam) {
        let osParamStr = osParam.toString().toLowerCase()
        if (oses.findIndex(os => os.value === osParamStr) >= 0)
          defaultSelectedOS = osParamStr
      } else {
        const userOS = detectOS()
        switch (userOS) {
          case UserOS.MAC:
            defaultSelectedOS = "mac"
            if (typeof document !== "undefined") {
              let gl = document.createElement("canvas").getContext("webgl")
              // @ts-ignore
              let ext = gl && gl.getExtension("WEBGL_debug_renderer_info")
              // @ts-ignore
              let param = (ext && ext.getParameter(d.UNMASKED_RENDERER_WEBGL)) || ""
              if (param.match(/Apple/) && !param.match(/Apple GPU/)) {
                defaultSelectedArch = "aarch64"
              }
            }
            break
          case UserOS.LINUX:
          case UserOS.UNIX:
            defaultSelectedOS = "linux"
            break
          default:
            defaultSelectedOS = "windows"
            break
        }
      }

      // init the default selected Package Type, if any from the param 'package'
      let defaultSelectedPackageType = defaultPackageType
      const packageParam = queryStringParams.package
      if (packageParam) {
        let packageParamStr = packageParam.toString().toLowerCase()
        if (packageTypes.findIndex(p => p.value === packageParamStr) >= 0)
          defaultSelectedPackageType = packageParamStr
      }

      // init the default selected Version, if any from the param 'version' or from 'variant'
      let defaultSelectedVersion = data.mostRecentLts.version
      const versionParam = queryStringParams.version
      if (versionParam) {
        let versionParamStr = versionParam.toString()
        let versionParamNum = Number(versionParamStr)

        if (versionParamStr.toLowerCase() === "latest") {
          // get the latest version of the list
          defaultSelectedVersion = data.allVersions.sort(
            (a, b) => b.node.version - a.node.version,
          )[0].node.version
        } else if (
          data.allVersions.edges.findIndex(
            version => version.node.version === versionParamNum,
          ) >= 0
        ) {
          defaultSelectedVersion = versionParamNum
        }
      }

      // init the default selected Version, if any from the param 'variant'
      const variantParam = queryStringParams.variant
      if (variantParam) {
        // convert openjdk11 to 11
        const parsedVersion = variantParam.toString().replace(/\D/g, "")
        let variantParamNum = Number(parsedVersion)

        if (data.allVersions.findIndex(version => version.node.version === variantParamNum) >= 0) {
          defaultSelectedVersion = variantParamNum
        }
      }

      osUpdater(defaultSelectedOS);
      archUpdater(defaultSelectedArch);
      versionUpdater(defaultSelectedVersion);

      // OK we can loaded elements
      setReady(true)
    })();
  }, [oses, arches]);

  /**
   * This useEffect() is called when a parameter is changed
   */
  useEffect(() => {
    // do nothing while params are not ready
    if(!ready) return

    (async () => {
      setReleases(
        await getAllPkgsForVersion(
          version,
          os,
          arch,
          packageType,
          selectedVendorIdentifiers,
        ),
      )
    })()
  }, [ready, version, os, arch, packageType, selectedVendorIdentifiers])

  const versionUpdater = version => {
    setURLParam("version", version)
    udateVersion(version)
  }

  const archUpdater = arch => {
    setURLParam("arch", arch)
    updateArch(arch)
  }

  const osUpdater = os => {
    setURLParam("os", os)
    updateOS(os)
  }

  return (
    <div className="max-w-[1264px] mx-auto px-6 pb-20">
      <ReleaseSelector
        marketplace
        versions={data.allVersions}
        updateVersion={versionUpdater}
        defaultVersion={version}
        updateOS={osUpdater}
        defaultOS={os}
        updateArch={archUpdater}
        defaultArch={arch}
        updatePackageType={updatePackageType}
      />

      <VendorSelector
        selectedVendorIdentifiers={selectedVendorIdentifiers}
        setSelectedVendorIdentifiers={updateSelectedVendorIdentifiers}
      />
      <div className="hidden md:flex items-center gap-20 border-t border-[#3E3355] pt-5 mb-5 mt-12">
        <p className="text-grey text-[16px] leading-[24px] mb-0">
          <Trans>Build Version</Trans>
        </p>
        <p className="text-grey text-[16px] leading-[24px] mb-0">
          <Trans>Distribution</Trans>
        </p>
        <p className="text-grey text-[16px] leading-[24px] mb-0">
          <Trans>Vendor</Trans>
        </p>
        <p className="text-grey text-[16px] leading-[24px] mb-0">Build Date</p>
        <p className="text-grey text-[16px] leading-[24px] mb-0">
          <Trans>Operating System</Trans>
        </p>
        <p className="text-grey text-[16px] leading-[24px] mb-0">
          <Trans>Architecture</Trans>
        </p>
      </div>
      {releases && <AllReleaseCard results={releases} />}
    </div>
  )
}

export default DownloadTable
