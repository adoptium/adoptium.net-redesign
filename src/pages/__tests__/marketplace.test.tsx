import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it, vi } from "vitest"
import { axe } from "vitest-axe"
import Marketplace, { Head } from "../marketplace"
import AxiosInstance from "axios"
import MockAdapter from "axios-mock-adapter"
import { mockOsesAPI, mockArchesAPI } from '../../__fixtures__/hooks'

const mock = new MockAdapter(AxiosInstance)

vi.mock('../../hooks/fetchConstants', () => {
  return {
    fetchOses: () => {
      // NOTE: we need to have the windows entry because it's the default detectedOS
      return [...mockOsesAPI(), {name: "windows", value: "windows"}];
    },
    fetchArches: () => {
      // NOTE: we need to have the x64 entry because it's the default detectedOS
      return [...mockArchesAPI(), {name: "x64", value: "x64"}];
    }
  };
})

vi.mock("../../util/shuffle", () => {
  return {
    shuffle: array => {
      array = [
        {
          name: "Adoptium",
          icon: "vendor-adoptium.png",
          postDownload: null,
        },
      ]
      return array
    },
  }
})

afterEach(() => {
  vi.clearAllMocks()
})

describe("Marketplace page", () => {
  it("renders correctly", () => {
    mock.onGet().reply(200, [])

    const { container } = render(<Marketplace />)
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")
    expect(pageContent).toMatchSnapshot()
  })

  it("head renders correctly", () => {
    mock.onGet().reply(200, [])

    const { container } = render(<Head />)
    // eslint-disable-next-line
    const title = container.querySelector("title")
    expect(title?.textContent).toEqual("Marketplace | Adoptium")
  })

  it("has no accessibility violations", async () => {
    mock.onGet().reply(200, [])

    const { container } = render(<Marketplace />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
