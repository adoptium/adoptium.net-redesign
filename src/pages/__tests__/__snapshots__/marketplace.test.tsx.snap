// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Marketplace page > renders correctly 1`] = `
<main>
  <div
    class="relative pt-40 pb-16 md:pb-36 md:pt-60 overflow-hidden"
  >
    <div
      class="w-full relative"
    >
      <div
        class="absolute z-[-1] w-[2396px] h-[1340px] left-[23px] md:left-[112px] top-[21px] md:top-[171px] bg-[#410170] shadow-[0_0_400px_rgba(65,1,112,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[1487px] h-[893px] left-[120px] md:left-[676px] top-[120px] md:top-[395px] bg-[#B62987] shadow-[0_0_400px_rgba(182,41,135,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[688px] h-[446px] left-[400px] md:left-[1131px] top-[618px] bg-[#FE8492] shadow-[0_0_400px_rgba(254,132,146,1)] rounded-full blur-[400px]"
      />
    </div>
    <div
      class="mx-auto max-w-[1048px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
    >
      <div
        class="self-stretch flex-col justify-center items-center gap-6 flex"
      >
        <div
          class="self-stretch  flex-col justify-center items-center gap-4 flex"
        >
          <div
            class="justify-start items-center gap-3 inline-flex"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#FF1464"
                height="8"
                rx="2"
                width="8"
                x="3"
                y="3"
              />
              <rect
                height="11"
                rx="3.5"
                stroke="#FF1464"
                stroke-opacity="0.25"
                stroke-width="3"
                width="11"
                x="1.5"
                y="1.5"
              />
            </svg>
            <div
              class="text-rose-600 text-base font-bold leading-normal"
            >
              Marketplace
            </div>
          </div>
          <div
            class="self-stretch text-center text-white   text-[56px] lg:text-[80px] leading-[114.286%] md:leading-[120%] font-semibold"
          >
            Adoptium® Marketplace
          </div>
        </div>
        <div
          class="self-stretch text-center text-lightgrey text-[20px] font-normal leading-[140%] undefined"
        >
          Java™ is the world's leading programming language and platform. The Adoptium Marketplace promotes high-quality, TCK certified and AQAvit verified runtimes for use across the Java ecosystem.
        </div>
      </div>
    </div>
  </div>
  <div
    class="max-w-[1264px] mx-auto px-6 pb-20"
  >
    <div>
       
      <div
        class="my-[50px]"
      >
        <div
          class="max-w-[1264px] mx-auto w-full"
        >
          <button
            class="sm:hidden flex justify-between items-center w-full text-[16px] font-normal leading-[24px] px-4 py-3 rounded-[80px] border-[2px] border-[#3E3355]"
          >
            Filter
            <span>
              <svg
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 4.16699V15.8337"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M4.16602 10H15.8327"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
          </button>
          <div
            class="w-full lg:overflow-visible overflow-auto hidden sm:block"
          >
            <div
              class="flex items-center gap-5 justify-between flex-nowrap min-w-[1039px]"
            >
              <div
                class="flex flex-col gap-4 w-full"
              >
                <h3
                  class="text-[16px] font-normal leading-[24px] text-grey"
                >
                  Operating System
                </h3>
                <div
                  class="relative mt-1"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="relative w-full flex items-center justify-between  rounded-[80px] border-[2px] bg-transparent py-3 pl-8 pr-4 border-[#3E3355]"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r0:"
                    type="button"
                  >
                    <span
                      class="flex items-center justify-between text-nowrap"
                    >
                      windows
                    </span>
                    <span>
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="flex flex-col gap-4 w-full"
              >
                <h3
                  class="text-[16px] font-normal leading-[24px] text-grey"
                >
                  Architecture
                </h3>
                <div
                  class="relative mt-1"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="relative w-full flex items-center justify-between  rounded-[80px] border-[2px] bg-transparent py-3 pl-8 pr-4 border-[#3E3355]"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r1:"
                    type="button"
                  >
                    <span
                      class="flex items-center justify-between text-nowrap"
                    >
                      x64
                    </span>
                    <span>
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="flex flex-col gap-4 w-full"
              >
                <h3
                  class="text-[16px] font-normal leading-[24px] text-grey"
                >
                  Version
                </h3>
                <div
                  class="relative mt-1"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="relative w-full flex items-center justify-between  rounded-[80px] border-[2px] bg-transparent py-3 pl-8 pr-4 border-[#3E3355]"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r2:"
                    type="button"
                  >
                    <span
                      class="flex items-center justify-between text-nowrap"
                    >
                      1 - LTS
                    </span>
                    <span>
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="flex flex-col gap-4 w-full"
              >
                <h3
                  class="text-[16px] font-normal leading-[24px] text-grey"
                >
                  Package Type
                </h3>
                <div
                  class="relative mt-1"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="relative w-full flex items-center justify-between  rounded-[80px] border-[2px] bg-transparent py-3 pl-8 pr-4 border-[#3E3355]"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r3:"
                    type="button"
                  >
                    <span
                      class="flex items-center justify-between text-nowrap"
                    >
                      JDK
                    </span>
                    <span>
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full"
    >
      <div
        class="overflow-auto min-w-full w-full"
      >
        <div
          class="flex xl:justify-center space-x-5 py-2"
        >
          <button
            class="flex justify-center items-center
                outline-none cursor-pointer transition-all duration-200 ease-in-out bg-white rounded-2xl border-primary border-2"
          >
            <span
              class="w-20 h-20"
            >
              <img
                alt="Adoptium icon"
                class="mb-0 p-1"
                src="/images/vendors/vendor-adoptium.png"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="hidden md:flex items-center gap-20 border-t border-[#3E3355] pt-5 mb-5 mt-12"
    >
      <p
        class="text-grey text-[16px] leading-[24px] mb-0"
      >
        Text
      </p>
      <p
        class="text-grey text-[16px] leading-[24px] mb-0"
      >
        Text
      </p>
      <p
        class="text-grey text-[16px] leading-[24px] mb-0"
      >
        Text
      </p>
      <p
        class="text-grey text-[16px] leading-[24px] mb-0"
      >
        Build Date
      </p>
      <p
        class="text-grey text-[16px] leading-[24px] mb-0"
      >
        Text
      </p>
      <p
        class="text-grey text-[16px] leading-[24px] mb-0"
      >
        Text
      </p>
    </div>
  </div>
  <div
    class="community flex justify-center items-center px-6 pt-10 pb-24 md:py-24"
  >
    <div
      class="newscard-2 gap-4 md:gap-10 flex flex-col md:flex-row justify-between md:items-center max-w-[1000px] mx-auto p-[32px] md:p-[64px]"
    >
      <div
        class="max-w-[110px] "
      >
        <img
          alt="AQAvit Logo"
          class="w-full"
          src="/images/projects/aqavit-project.svg"
        />
      </div>
      <div
        class="flex flex-col md:gap-3"
      >
        <div
          class="!mt-0"
        >
          <h2
            class="text-4xl leading-[122%] md:text-5xl md:[116%] text-white font-semibold"
          >
            About Eclipse AQAvit
          </h2>
          <p
            class="text-grey mt-6 font-normal text-base"
          >
            The AQAvit open source test suite (Adoptium Quality Assurance) can be found here. There is also a blog post and brief presentation that explains what testing is run and how it fits into the overall delivery pipeline.
          </p>
        </div>
        <button
          class="bg-transparent mt-10 border-2 border-pink-500/0 text-white text-base  leading-6 font-normal w-[146px] h-[48px] rounded-[80px] gradient-border mt-6"
        >
          Learn More
        </button>
      </div>
    </div>
  </div>
</main>
`;
